packages/labs/react/src/create-component.ts

```diff
@@ -84,16 +84,32 @@ const setRef = (ref: React.Ref<unknown>, value: Element | null) => {
   }
 };
 
-type Events<S> = {
-  [P in keyof S]?: (e: Event) => unknown;
-};
-
 type StringValued<T> = {
   [P in keyof T]: string;
 };
 
 type Constructor<T> = {new (): T};
 
+type EventString<T extends Event = Event> = string & {
+  __event_type: T;
+};
+
+/***
+ * Typecast that curries an Event type through an EventString. The React
+ * equivelant to an event listeners is an `EventName`. The goal of the type
+ * name is to hint that corresponding EventName callback in their props will
+ * be of a certain type.
+ */
+export type EventName<T> = T extends Event ? EventString<T> : never;
```

Comment:
I don't think you don't need this type at all. Once you have the type constraint of `T extends Event` TypeScript will fail for any type parameter that doesn't extend `Event` so you could never take the else clause here. I tried this locally and it works fine.

Remove this type, rename `EventString` to `EventName`, and export that.

Reply:
That makes sense :) very cool!

Reply:
done!